client/my-sites/checkout/composite-checkout/components/wp-checkout.js

```diff
@@ -102,6 +104,47 @@ const OrderReviewTitle = () => {
 
 const paymentMethodStep = getDefaultPaymentMethodStep();
 
+// Can be safely removed after 2021-02-14 when the FRESHPACK coupon expires
+const useFRESHPACKPromoCodePrefill = ( applyCoupon ) => {
+	const done = useRef();
+	const [ items ] = useLineItems();
+
+	useEffect( () => {
+		// Don't fully execute more than once
+		if ( done.current ) {
+			return;
+		}
+
+		// Only ever pre-fill the FRESHPACK coupon for people
+		// who are part of the corresponding A/B test group
+		const shouldPrefill = abtest( 'prefillFRESHPACKCouponCode' ) === 'test';
+		if ( ! shouldPrefill ) {
+			done.current = true;
+			return;
+		}
+
+		// We don't always immediately have a list of products in the cart;
+		// if the cart doesn't contain any products in its line items,
+		// don't go any further until they're loaded
+		const productSlugs = items
```

Comment:
I think that it might be wise to go directly to the shopping cart here rather than `useLineItems`.

So above you could do `const { responseCart } = useShoppingCart();`, then here do `const productSlugs = responseCart.products.map( product => product.product_slug );`